### ISSUE TITLE
crash seen while attaching files by drag and drop 

### ISSUE THREAD

**Issue by yuvarajgogoi** [InvictusInnovations/keyhotee#277]
Steps to reproduce:
1) open a client
2) compose 
3) try to attach a large no of files to attach by drag and drop.I tried with 22 text files.

crash is seen:

Tested in Ubuntu 13.10 and is always reproducible.

log : 

927785ms       th_a              startup ] .            KeyhoteeApplication.cpp:286
927786ms       th_a         get_profiles ] profiles []          application.cpp:390
927786ms       th_a displayProfileWizard ] .            KeyhoteeApplication.cpp:240
945447ms       th_a        createProfile ] .            ProfileWizard.cpp:256
945450ms       th_a               create ] Creating TAutoUpdateProgressBar          AutoUpdateProgressBar.cpp:73
945451ms       th_a               doTask ] Entering...          AutoUpdateProgressBar.cpp:84
945451ms       th_a               doTask ] Leaving...           AutoUpdateProgressBar.cpp:107
945452ms       th_a           operator() ] before mainTask()            AutoUpdateProgressBar.cpp:89
945452ms       th_a       create_profile ] config_file: /home/yuvaraj/.local/share/Keyhotee/profiles/Y G/config.json            application.cpp:419
945655ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
945679ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
945703ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
945726ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
945751ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
945774ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
945798ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
945822ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
945846ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
945869ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
945893ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
945917ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
945941ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
945964ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
945987ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946011ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946035ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946059ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946082ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946107ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946131ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946156ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946181ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946205ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946230ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946255ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946280ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946305ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946328ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946353ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946378ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946403ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946428ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946451ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946477ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946501ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946527ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946551ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946575ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946600ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946626ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946651ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946676ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946700ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946724ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946749ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946775ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946800ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946825ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946848ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946873ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946899ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946924ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946949ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946972ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
946997ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947022ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947047ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947072ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947096ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947120ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947146ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947172ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947197ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947222ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947246ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947270ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947296ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947321ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947345ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947369ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947394ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947419ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947444ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947469ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947492ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947517ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947541ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947565ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947589ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947612ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947636ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947660ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947684ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947707ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947731ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947755ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947778ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947802ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947826ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947849ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947873ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947896ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947920ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947943ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947967ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
947990ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
948015ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
948038ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
948062ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
948085ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
948109ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
949149ms  stretch_s          updateValue ] Updating progress value...           AutoUpdateProgressBar.cpp:118
949167ms       th_a               create ] finished stretch_seed            profile.cpp:84
949167ms       th_a                 open ] opening profile: /home/yuvaraj/.local/share/Keyhotee/profiles/Y G            profile.cpp:99
949168ms       th_a                 open ] loading master key file:.keyhotee_master.key         profile.cpp:112
949168ms       th_a                 open ] opening profile databases            profile.cpp:124
950882ms       th_a                 open ] finished opening profile         profile.cpp:138
950883ms       th_a            configure ] application::configure           application.cpp:219
950883ms       th_a            configure ] calling get_external_ip          application.cpp:226
950883ms       th_a      get_external_ip ] resolve ipecho.net port 80           ipecho.cpp:20
954848ms       th_a      get_external_ip ] ip line=106.51.196.129
            ipecho.cpp:44
954849ms       th_a            configure ] external IP 106.51.196.129           application.cpp:228
954849ms       th_a            configure ] set_external_ip          application.cpp:237
954849ms       th_a            configure ] configuring server           application.cpp:250
954849ms       th_a            configure ] listening for stcp connections on port 0         server.cpp:207
954849ms       th_a            configure ] configure bitname client         application.cpp:256
955258ms       th_a         load_indexes ] load indexes         bitname_db.cpp:92
955258ms       th_a         load_indexes ] load indexes         bitname_db.cpp:107
955261ms       th_a      init_timekeeper ] ...init stats...         bitname_db.cpp:162
955261ms       th_a      init_timekeeper ] ...done init timekeeper...           bitname_db.cpp:164
955261ms       th_a                 open ] open name db         bitname_db.cpp:214
956146ms       th_a            configure ] configure bitchat client         application.cpp:266
956447ms       th_a            configure ] configuring rpc_server           application.cpp:270
956447ms       th_a            configure ] listening for rpc connections on port 0          rpc_server.cpp:245
956447ms       th_a            configure ] listening for rpc connections on port 46988          rpc_server.cpp:247
956448ms       th_a            configure ] done configuring rpc_server          application.cpp:273
956448ms       th_a            configure ] end application::configure           application.cpp:275
956448ms       th_a           operator() ] after mainTask()         AutoUpdateProgressBar.cpp:91
956448ms       th_a             onFinish ] Entering...          AutoUpdateProgressBar.cpp:136
956448ms       th_a             onFinish ] executing _onFinishAction            AutoUpdateProgressBar.cpp:139
956563ms       th_a   KeyhoteeMainWindow ] idents: []           KeyhoteeMainWindow.cpp:272
956595ms       th_a start_mail_connect_l ] start_mail_connect_loop          application.cpp:154
956595ms       th_a             onFinish ] Leaving...           AutoUpdateProgressBar.cpp:141
956595ms       th_a           operator() ] after notifyFinished()           AutoUpdateProgressBar.cpp:93
961599ms       th_a    mail_connect_loop ] mail connect 162.243.67.4:7896           application.cpp:80
961600ms       th_a              connect ] connect to 162.243.67.4:7896 and resolved ["162.243.67.4:7896"]          mail_connection.cpp:208
962036ms       th_a              connect ]     connected to 162.243.67.4:7896           mail_connection.cpp:199
967468ms       th_a        store_contact ] to_string()          addressbook.cpp:110
967469ms       th_a        store_contact ] ... did it work?         addressbook.cpp:112
981533ms       th_a   displayFailureInfo ] Unknown exception caught         KeyhoteeApplication.cpp:259
981533ms       th_a   displayFailureInfo ] fatal error Unknown exception caught         KeyhoteeApplication.cpp:261

**Comment by PaulEU** [comment]
I can't reproduce it on windows 7.
I attached 250 files, size of all files about: 40 GB size.
Please check Yuvarajgogoi. Do you also have this error when you use "paste" option and "add attachment" option ?
Mayby we should  check max. message size before adding to attachments table ?

**Comment by yuvarajgogoi** [comment]
I got a crash again in the same scenario with 92 files now. I selected all files togather and dropped at the attachment widget. It crashed. tested in the latest sources. 

add me at skype  y u v a r a j . g o g o i  (without space) if you want more info. 

3031446ms       th_a              startup ] .           KeyhoteeApplication.cpp:286
3031447ms       th_a         get_profiles ] /home/yuvaraj/.local/share/Keyhotee/profiles/Y G            application.cpp:383
3031447ms       th_a         get_profiles ] /home/yuvaraj/.local/share/Keyhotee/profiles/J G            application.cpp:383
3031447ms       th_a         get_profiles ] profiles ["Y G","J G"]          application.cpp:390
3031447ms       th_a         displayLogin ] .           KeyhoteeApplication.cpp:216
3031498ms       th_a         get_profiles ] /home/yuvaraj/.local/share/Keyhotee/profiles/Y G            application.cpp:383
3031498ms       th_a         get_profiles ] /home/yuvaraj/.local/share/Keyhotee/profiles/J G            application.cpp:383
3031498ms       th_a         get_profiles ] profiles ["Y G","J G"]          application.cpp:390
3031498ms       th_a          LoginDialog ] profiles Y G            LoginDialog.cpp:34
3031498ms       th_a          LoginDialog ] profiles J G            LoginDialog.cpp:34
3037419ms       th_a                 open ] opening profile: /home/yuvaraj/.local/share/Keyhotee/profiles/Y G           profile.cpp:99
3037419ms       th_a                 open ] loading master key file:.keyhotee_master.key            profile.cpp:112
3037435ms       th_a                 open ] Could not open .keyhotee_master.key, trying to open legacy key file (.strecthed_seed).          profile.cpp:120
3037435ms       th_a              onLogin ] error assert
fc::exists( file ): 
    {}
    th_a  aes.cpp:295 aes_load

```
{"file":"/home/yuvaraj/.local/share/Keyhotee/profiles/Y G/.stretched_seed"}
th_a  aes.cpp:311 aes_load

{"profile_dir":"/home/yuvaraj/.local/share/Keyhotee/profiles/Y G"}
th_a  profile.cpp:139 open

{}
th_a  application.cpp:312 loadProfile           LoginDialog.cpp:72
```

3042642ms       th_a                 open ] opening profile: /home/yuvaraj/.local/share/Keyhotee/profiles/Y G           profile.cpp:99
3042642ms       th_a                 open ] loading master key file:.keyhotee_master.key            profile.cpp:112
3042643ms       th_a                 open ] opening profile databases           profile.cpp:124
3043814ms       th_a                 open ] finished opening profile            profile.cpp:138
3043827ms       th_a            configure ] application::configure          application.cpp:219
3043827ms       th_a            configure ] calling get_external_ip         application.cpp:226
3043827ms       th_a      get_external_ip ] resolve ipecho.net port 80          ipecho.cpp:20
3044461ms       th_a      get_external_ip ] ip line=106.51.180.25
            ipecho.cpp:44
3044462ms       th_a            configure ] external IP 106.51.180.25           application.cpp:228
3044462ms       th_a            configure ] set_external_ip         application.cpp:237
3044462ms       th_a            configure ] configuring server          application.cpp:250
3044462ms       th_a            configure ] listening for stcp connections on port 9876         server.cpp:207
3044462ms       th_a            configure ] configure bitname client            application.cpp:256
3044795ms       th_a         load_indexes ] load indexes            bitname_db.cpp:92
3044795ms       th_a         load_indexes ] load indexes            bitname_db.cpp:107
3044816ms       th_a      init_timekeeper ] ...init stats...            bitname_db.cpp:162
3044816ms       th_a      init_timekeeper ] ...done init timekeeper...          bitname_db.cpp:164
3044816ms       th_a                 open ] open name db            bitname_db.cpp:214
3045326ms       th_a            configure ] configure bitchat client            application.cpp:266
3045539ms       th_a            configure ] configuring rpc_server          application.cpp:270
3045539ms       th_a            configure ] listening for rpc connections on port 9877          rpc_server.cpp:245
3045539ms       th_a            configure ] listening for rpc connections on port 9877          rpc_server.cpp:247
3045539ms       th_a            configure ] done configuring rpc_server         application.cpp:273
3045539ms       th_a            configure ] end application::configure          application.cpp:275
3045701ms       th_a   KeyhoteeMainWindow ] idents: []          KeyhoteeMainWindow.cpp:272
3045714ms       th_a start_mail_connect_l ] start_mail_connect_loop         application.cpp:154
3050714ms       th_a    mail_connect_loop ] mail connect 162.243.67.4:7896          application.cpp:80
3050715ms       th_a              connect ] connect to 162.243.67.4:7896 and resolved ["162.243.67.4:7896"]         mail_connection.cpp:208
3051199ms       th_a              connect ]     connected to 162.243.67.4:7896          mail_connection.cpp:199
3053393ms       th_a        store_contact ] to_string()         addressbook.cpp:110
3053393ms       th_a        store_contact ] ... did it work?            addressbook.cpp:112
3097937ms       th_a   displayFailureInfo ] Unknown exception caught            KeyhoteeApplication.cpp:259
3097937ms       th_a   displayFailureInfo ] fatal error Unknown exception caught            KeyhoteeApplication.cpp:261
